qcodes/dataset/data_set_cache.py

```diff
@@ -252,7 +294,20 @@ def _insert_into_data_dict(
         shape: Optional[Tuple[int, ...]]
 ) -> Tuple[np.ndarray, Optional[int]]:
     if shape is None or write_status is None:
-        return np.append(existing_values, new_values, axis=0), None
+        try:
+            data = np.append(existing_values, new_values, axis=0)
+        except ValueError:
+            # we cannot append into a ragged array so make that manually
+            n_existing = existing_values.shape[0]
+            n_new = new_values.shape[0]
+            n_rows = n_existing + n_new
+            data = np.ndarray((n_rows,), dtype=object)
```

Comment:
why is the type of the existing values not re-used? or will numpy do it automatically behind the scenes?
```suggestion
            data = np.ndarray((n_rows,), dtype=existing_values.dtype)
```

Reply:
Because this is a ragged array as written in the comment.

Reply:
ah, indeed.